File: CA01/calculator.py
"""
B8IT105 Programming for Big Data
Assignment 1 - A 10 Function Calculator

Submitted by:
Deirdre Flaherty (10349680)
"""

import math

class Calculator(object):

    #Function1: add two numbers together
    def add(self, first, second):
        number_types = (int, long, float, complex)
        if isinstance(first, number_types) and isinstance(second, number_types):
            return first + second
        else:
            return 'error'
        
    #Function2: subtract one number from another
    def subtract(self, first, second):
        number_types = (int, long, float, complex)
        if isinstance(first, number_types) and isinstance(second, number_types):
            return first- second
        else:
            return 'error'
    
    #Function3: multiply two numbers
    def multiply(self, first, second):
        number_types = (int, long, float, complex)
        if isinstance(first, number_types) and isinstance(second, number_types):
            return first * second
        else:
             return 'error'
    
    #Function4: divide one number by another
    def divide(self, first, second):
        number_types = (int, long, float, complex)
        if isinstance(first, number_types) and isinstance(second, number_types):
            if second == 0:
                return 'undefined'
            else:
                return first / float(second)
        else:
            return 'error'
    
    #Function5: exponentiate one number by another
    def exponent(self, first, second):
        number_types = (int, long, float, complex)
        if isinstance(first, number_types) and isinstance(second, number_types):
            return first ** second
        else:
            return 'error'
        
    #Function6: sqare a number
    def square(self, num):
        number_types = (int, long, float, complex)
        if isinstance(num, number_types):
            return num ** 2
        else:
            return 'error'
            
    #Function7: cube a number
    def cube(self, num):
        number_types = (int, long, float, complex)
        if isinstance(num, number_types):
            return num ** 3
        else:
            return 'error'
            
    #Function8: get the sqare root of a number (exponentiate by 0.5)
    def squareroot(self, num):
        number_types = (int, long, float, complex)
        if isinstance(num, number_types):
            #negative numbers cannot be raised to a fractional power
            if num < 0:
               return 'error'
            else:
                return num ** 0.5
        else:
            return 'error'
    
    #Function9: return the sine of x radians
    def sine(self, x):
        number_types = (int, long, float, complex)
        if isinstance(x, number_types):
            return math.sin(x)
        else:
            return 'error'
    
    #Function10: return the cosine of x radians
    def cosine(self, x):
        number_types = (int, long, float, complex)
        if isinstance(x, number_types):
            return math.cos(x)
        else:
            return 'error'
    
    def printMenu(self):
        print "Please select an operation from the menu below:"
        print "1: Add"
        print "2: Subtract"
        print "3. Multiply"
        print "4. Divide"
        print "5. Exponent"
        print "6. Square"
        print "7. Cube"
        print "8. Square Root"
        print "9. Sine"
        print "10. Cosine"
        print "Enter Q to quit"
        
        
        


    

File: CA05/calculator.R
####################################################################################
#                           Calculator Functions
####################################################################################

# Function1: add two numbers together
add_two_numbers <- function(num1, num2){ 
  if (is.numeric(num1) & is.numeric(num2))
  {
    num1 + num2 
  } else {
    'error'
  }
}

# Function2: subtract one number from another
subtract_two_numbers <- function(num1, num2){ 
  if (is.numeric(num1) & is.numeric(num2))
  {
    num1 - num2 
  } else {
    'error'
  }
}

# Function3: multiply two numbers
multiply_two_numbers <- function(num1, num2){ 
  if (is.numeric(num1) & is.numeric(num2))
  {
    num1 * num2 
  } else {
    'error'  
  }
}

# Function4: divide one number by another
divide_two_numbers <- function(num1, num2){ 
  if (is.numeric(num1) & is.numeric(num2)) {
    if (num2 == 0) {
      'undefined' 
    } else {
      num1 / num2
    }
  } else {
    'error'
  }
}

# Function5: exponentiate one number by another
exponentiate_two_numbers <- function(num1, num2){ 
  if (is.numeric(num1) & is.numeric(num2))
  {
    num1 ** num2
  } else {
    'error'  
  }
}

#Function6: square a number
my_square <- function(num){ 
  if (is.numeric(num))
  {
    num ** 2
  } else {
    'error'  
  }
}

# Function7: cube a number
my_cube <- function(num){ 
  if (is.numeric(num))
  {
    num ** 3
  } else {
    'error'  
  }
}

# Function8: get the sqare root of a number (exponentiate by 0.5)
my_sqrt <- function(num){ 
  if (is.numeric(num) & num > 0)
  {
    num ** 0.5
  } else {
    'error'  
  }
}

# Function9: return the sine of x radians
my_sin <- function(num){
  if (is.numeric(num))
  {
    sin(num)
  } else {
    'error'  
  }
}

# Function10: return the cosine of x radians
my_cos <- function(num){
  if (is.numeric(num))
  {
    cos(num)
  } else {
    'error'  
  }
}

####################################################################################
#                           Calls
####################################################################################

# Test add_two_numbers function
add_two_numbers(5,3)
add_two_numbers(4,0)
add_two_numbers('fred',0)

# Test subtract_two_numbers function
subtract_two_numbers(5,3)
subtract_two_numbers(4,0)
subtract_two_numbers(3,4)
subtract_two_numbers('fred',4)

# Test multiply_two_numbers function
multiply_two_numbers(5,3)
multiply_two_numbers(4,0)
multiply_two_numbers(3,-4)
multiply_two_numbers(3,'fred')

# Test divide_two_numbers function
divide_two_numbers(6,2)
divide_two_numbers(6,0)
divide_two_numbers(6,-2)
divide_two_numbers(6,'fred')

# Test exponentiate_two_numbers function
exponentiate_two_numbers(5,3)
exponentiate_two_numbers(4,1)
exponentiate_two_numbers(4,0)
exponentiate_two_numbers(2,-2)
exponentiate_two_numbers(2,'fred')

# Test my_square function
my_square(3)
my_square(5)
my_square(-3)
my_square('fred')

# Test my_cube function
my_cube(2)
my_cube(5)
my_cube(-2)
my_cube('fred')

# Test my_sqrt function
my_sqrt(9)
my_sqrt(25)
my_sqrt(5)
my_sqrt(-4)
my_sqrt('fred')

# Test my_sine function
my_sin(3)
my_sin(-3)
my_sin(0)
my_sin('fred')

# Test my_cos function
my_cos(0)
my_cos(3)
my_cos(-3)
my_cos('fred')
